### DISCUSSION TITLE
Behavior when a game is uninstalled (linux)

### DISCUSSION THREAD

**Discussion by GameKing505** [mtkennerly/ludusavi#421] [Q&A] [answered]
Hello,

I've recently discovered the ludusavi tool and it seems like exactly what I need, as a gamer on linux. My one question is - what is the expected behavior regarding treatment of backed up files after the game is uninstalled and the source folder no longer exists?

On linux it is often the case that when I game is uninstalled, the entire proton prefix is deleted (which contains the save files), so I want to clarify what happens when games are uninstalled. Let's say for example:

- I run my initial backup, which among other games picks up my Dark Souls 2 save. That backup is also pushed to the cloud via the rclone integration.
- I have automated backups run every day at midnight via systemd
- At some point, I am done playing DS2. At this point I uninstall it.
- Midnight on the next day, the ludusavi service runs again. 

On this last run - does ludusavi know to keep the Dark Souls 2 save, even if the source files are no longer there? Or will it detect that there was a deletion, and then propagate that deletion to the save archive? I have run a test and when I "preview" the changes it does seem to detect that the Dark Souls 2 files were deleted but I'm unsure what it does with that information.

Thanks for the help.

**Comment by mtkennerly** [comment, marked as answer]
Hi! The behavior will depend on exactly how much is removed by the uninstallation:

* If all the saves that Ludusavi had backed up are removed, then Ludusavi will ignore it and leave the existing backups alone.
* If *some* of the saves are removed, but not all, then Ludusavi will detect that as a change and make a new backup without those removed files.

In the latter case, one option you have is to lock the most recent backup, which means that that backup will be kept indefinitely even when new backups are created (restore screen -> game three-dot menu). You could also disable the game in Ludusavi (backup screen) to prevent it from making any new backups.

**Comment by GameKing505** [comment]
This is very helpful - thank you for clarifying. In the Dark Souls 2 example, the entire prefix is deleted so it seems the save file will be safe.

I will say the UI is not super clear about this - it shows that the screenshots folder for DS2 will be backed up but the actual save files are no longer listed in the dropdown. Not sure what a good solution there is but at least I know the right stuff is happening behind the scenes.

**Reply by mtkennerly** [reply]
If the saves aren't listed, then that means it's already created a backup without them, unfortunately. When they've been deleted on your system but not yet removed from the backup, then Ludusavi would list them with an X icon.

**Reply by mtkennerly** [reply]
I could add an option so that Ludusavi won't make a new backup if there are only removed saves and no new/modified saves. I think that would help here. Would you be interested in that?

**Reply by GameKing505** [reply]
Yes! I think that would be very helpful. Sort of an "add-only" mode so the user doesn't need to worry about deletions propagating.

**Reply by mtkennerly** [reply]
@GameKing505 Just wanted to let you know that the next release will include an option for this :) I'm planning to publish the release within a week or so.

> <img width="433" alt="image" src="https://github.com/user-attachments/assets/56f97c62-c54c-46dd-88be-dab74292fe8c" />

**Reply by GameKing505** [reply]
Amazing!
